winml/api/Windows.AI.MachineLearning.idl

```diff
@@ -16,9 +16,18 @@ import "windows.graphics.imaging.idl";
 import "windows.storage.idl";
 #endif
 
+#ifndef ROOT_NS
+#define ROOT_NS Windows
+#endif
+
+#define STRINGIFY(x) #x
+#define XSTRINGIFY(x) STRINGIFY(x)
```

Comment:
what does his one do? looks like it just passed the arg through

Reply:
macros will expand variables when they are passed into functions, but not #. so

#define foo bar
STRINGIFY(foo) = #foo = "foo"
XSTRINGIFY(foo) = STRINGIFY(bar) = "bar"